### PR TITLE
ejs/more report filters

### PR DESCRIPTION
**Pull request by jmtroth0**
so far this is what I've worked through:

corehq/apps/reports/templates/reports/filters/drillable_async.html 3b126d6 --> to custom 
corehq/apps/reports/templates/reports/filters/location_async.html --> 002d5d6  
corehq/apps/reports/templates/reports/filters/drilldown_options.html --> ff16ac5  
corehq/apps/reports/templates/reports/filters/filter_users.html 41b1b6d
corehq/apps/reports/templates/reports/filters/devicelog_tags.html --> ab2a8a8

@orangejenny / @mkangia

### CODE REVIEW

**Comment by jmtroth0** [comment]
(for a bit, the diff will look weird since I merged in #20167 without waiting for it to be merged into master)

**Review by mkangia — COMMENTED** [review]
@jmtroth0 
Is this work from multiple related PRs/discussions.. i dont have the context to review this to be fairly honest. If its not so and i can review this, can you add more info on what this PR is aiming to achieve..

**Comment by jmtroth0** [comment]
@mkangia yeah, this is a last push towards externalizing javascript. There are no functional changes included, but it is definitely a tough pr to review. Commit by commit should be decent if you want to take a stab at it.

@orangejenny This (absurd) pr now has all of the final filters

**Comment by jmtroth0** [comment]
I just merged in #20176, so again the diff looks weird until that is merged

File: corehq/apps/reports/filters/fixtures.py
Comment (orangejenny) [inline corehq/apps/reports/filters/fixtures.py]: the best of intentions...

File: corehq/apps/reports/static/reports/js/report_filter.drilldown_options.js
Comment (orangejenny) [inline corehq/apps/reports/static/reports/js/report_filter.drilldown_options.js]: When changing from classical to functional inheritance, the naming convention switches from ucfirst to lcfirst - https://github.com/dimagi/commcare-hq/pull/19895#pullrequestreview-106915955

File: corehq/apps/reports/templates/reports/filters/drilldown_options.html
Comment (orangejenny) [inline corehq/apps/reports/templates/reports/filters/drilldown_options.html]: FYI, there's a `{% html_attr option_map %}` that serializes as JSON and also does HTML attribute escaping. I think JSON escape single quotes so you're fine here and it's `data-foo="{{ foo|JSON }}"` that's the risk.

File: corehq/apps/reports/templates/reports/filters/devicelog_tags.html
Comment (orangejenny) [inline corehq/apps/reports/templates/reports/filters/devicelog_tags.html]: Can delete the script tags around this.

File: corehq/apps/reports/static/reports/js/report_filter.drilldown_options.js
Comment (jmtroth0) [reply corehq/apps/reports/static/reports/js/report_filter.drilldown_options.js]: ooh nice, hadn't seen that convo ^^

File: corehq/apps/reports/templates/reports/filters/drilldown_options.html
Comment (jmtroth0) [reply corehq/apps/reports/templates/reports/filters/drilldown_options.html]: Yeah, it was odd at first that double quotes didn't work, but I figured that was the reason

File: corehq/apps/reports/static/reports/js/filters/main.js
Comment (orangejenny) [inline corehq/apps/reports/static/reports/js/filters/main.js]: I had imagined this id being a data attribute, since it's passed from python along with `default_on` and `errors_only`, though I"m not sure it matters much either way.

File: corehq/apps/reports/static/reports/js/filters/main.js
Comment (orangejenny) [inline corehq/apps/reports/static/reports/js/filters/main.js]: Nitpick and you didn't introduce this, but for clarity can you rename this to "advancedFormsOptions" with a d?

File: corehq/apps/reports_core/static/reports_core/js/base_template_new.js
Comment (orangejenny) [inline corehq/apps/reports_core/static/reports_core/js/base_template_new.js]: You tested these changes locally with both UCRs and standard reports, right?

File: corehq/apps/reports/static/reports/js/filters/main.js
Comment (jmtroth0) [reply corehq/apps/reports/static/reports/js/filters/main.js]: heh, already did generally, but i guess i missed this spot

File: corehq/apps/reports_core/static/reports_core/js/base_template_new.js
Comment (jmtroth0) [reply corehq/apps/reports_core/static/reports_core/js/base_template_new.js]: Yup. It worked on my reports, but it might be worth pulling and checking out more complex ones if you have them

Comment (orangejenny) [reply corehq/apps/reports_core/static/reports_core/js/base_template_new.js]: Cool, that sounds like enough due diligence.

File: corehq/apps/reports/templates/reports/filters/form_app_module_drilldown.html
Comment (orangejenny) [inline corehq/apps/reports/templates/reports/filters/form_app_module_drilldown.html]: I think this comment might be worth preserving, after looking back at the history (http://manage.dimagi.com/default.asp?231773)

File: corehq/apps/reports/static/reports/js/filters/main.js
Comment (orangejenny) [inline corehq/apps/reports/static/reports/js/filters/main.js]: Shouldn't this be `$el.find('#' + data.cssId + '-advanced-options')`?

**Review by orangejenny — APPROVED** [review]

**Pull request merged**
